### PR TITLE
Add keywords D1

### PR DESCRIPTION
**Pull request by stefoss23**
Adding more keywords beginning w/ D.

### CODE REVIEW

**Comment by stefoss23** [comment]
jenkins build this please

File: src/opm/parser/eclipse/share/keywords/keyword_list.cmake [hunk 1]
```diff
@@ -9,6 +9,25 @@
 # you are therefore encouraged to go to the opm-simulators repository and update
 # the file: opm/autodiff/MissingFeatures.cpp
 
+#Some keywords are found to be of 'special' structure:
+#W             WHAT
+#ECONT         Terminates with two /
+#COALNUM       matrix structure
+#COMPDATM      alias for COMPDATL
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/share/keywords/keyword_list.cmake]: You should only add one json file - with an *arbitrary name*, and then include the `deck_names` attribute:
```json
"deck_names":  ["COMPDATM","COMPDATL"], 
```
 see e.g. the json file [`W/WELL_PROBE`](https://github.com/OPM/opm-common/blob/master/src/opm/parser/eclipse/share/keywords/000_Eclipse100/W/WELL_PROBE)

File: src/opm/parser/eclipse/share/keywords/keyword_list.cmake [hunk 2]
```diff
@@ -9,6 +9,25 @@
...
+#COMPINJK      undef default
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/share/keywords/keyword_list.cmake]: This looks clear enough to me - they all have default value 0 - the "Allow any value" functionality will be handeled in the subsequent C++ layer.

File: src/opm/parser/eclipse/share/keywords/keyword_list.cmake [hunk 3]
```diff
@@ -9,6 +9,25 @@
...
+#COMPRP        strange defaults
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/share/keywords/keyword_list.cmake]: The strange defaults in `COMPRP` are "OK"; no default can be assigned in json, and the C++ code will have to look like this:
```
if item.defaultApplied(0)
    value = do_some_fancy_calculation_to_get_a_usable_value();
```

this is OK - we have many such keywords.

File: src/opm/parser/eclipse/share/keywords/keyword_list.cmake [hunk 4]
```diff
@@ -9,6 +9,25 @@
...
+#COMPRPL       strange defaults
+#COMPVE        default: negative
+#COMPVEL       default: negative
+#CSKIN         strange default
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/share/keywords/keyword_list.cmake]: The `CSKIN` looks smooth to me?

File: src/opm/parser/eclipse/share/keywords/keyword_list.cmake [hunk 5]
```diff
@@ -9,6 +9,25 @@
...
+#DEPTHTAB      different unit in table
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/share/keywords/keyword_list.cmake]: DEPTHTAB looks tricky - I would just go for `"SIZE_TYPE":"ALL"` and `"value_type":"STRING"` - i.e. completely ignore all unit conversion.

File: src/opm/parser/eclipse/share/keywords/keyword_list.cmake [hunk 6]
```diff
@@ -9,6 +9,25 @@
...
+#DIFFCOAL      default for item3 same as value for item 1
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/share/keywords/keyword_list.cmake]: Just do not assign a default value at all for item3.

File: src/opm/parser/eclipse/share/keywords/keyword_list.cmake [hunk 7]
```diff
@@ -9,6 +9,25 @@
...
+#DIFFMR-       no support for character '-' in kw. 
+#DIFFMTH-      no support for character '-' in kw. 
+#DIFFMX-       no support for character '-' in kw. 
+#DIFFMY-       no support for character '-' in kw.
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/share/keywords/keyword_list.cmake]: Define a json file `DIFF` with attribute `"deck_name":["DIFFMR-", "DIFFMR", "DIFFMTH-", "DIFFMX", "DIFFMX-", ...`

**Comment by stefoss23** [comment]
jenkins build this please

**Pull request merged**